tests/conftest.py

```diff
@@ -82,8 +82,12 @@ def snapshot(request):
         else:
             token = request_token(request).replace(" ", "_").replace(os.path.sep, "_")
 
-        with _snapshot_context(token, *snap.args, **snap.kwargs) as snapshot:
-            yield snapshot
+        mgr = _snapshot_context(token, *snap.args, **snap.kwargs)
+        snapshot = mgr.__enter__()
+        yield snapshot
+        # Skip doing any checks if the test was skipped
+        if hasattr(request.node, "rep_call") and not request.node.rep_call.skipped:
```

Comment:
Shouldn't this be `if hasattr(request.node, "rep_call") and request.node.rep_call.skipped`?

Reply:
the `__exit__` is what invokes the snapshot test logic. We only want to invoke it if the test isn't skipped

Reply:
I see, `__exit__()` continues `snapshot_context()` execution after the `snapshot_context()` yields (at this point, it has only submitted traces to the snapshot agent and has not made any assertions on the testagent's response).